okonomiyaki/platforms/epd_platform.py

```diff
@@ -88,26 +100,140 @@ def from_epd_string(cls, s):
+
+    def __init__(self, platform, **kw):
+        if not self._is_supported(platform):
+            msg = "Platform {0} not supported".format(platform)
+            raise OkonomiyakiError(msg)
+        super(EPDPlatform, self).__init__(platform=platform, **kw)
+
+    def _is_supported(self, platform):
+        arch_and_machine_are_intel = (
+            platform.arch in (_X86, _X64_64)
+            and platform.machine in (_X86, _X64_64)
+        )
+        if platform.os == WINDOWS:
+            return arch_and_machine_are_intel
+        if platform.os == DARWIN:
+            return arch_and_machine_are_intel
+        if platform.os == SOLARIS:
+            return arch_and_machine_are_intel
+        if platform.os == LINUX:
+            if platform.family != RHEL:
+                return False
+            parts = platform.release.split(".")
+            return parts[0] in ("3", "5", "6", "7") \
+                and arch_and_machine_are_intel
+
+        return False
+
+    @property
+    def arch(self):
+        return self.platform.arch
 
     @property
     def arch_bits(self):
         """
         The number of bits (as a string) of this epd platform.
         """
-        if self.arch == "x86":
-            return "32"
+        return self.arch._arch_bits
+
+    @property
+    def pep425_tag(self):
+        msg = "Cannot guess platform tag for platform {0!r}"
+
+        if self.platform.family == MAC_OS_X:
+            if self.platform.arch.name == X86:
+                return "macosx_10_6_i386"
+            elif self.platform.arch.name == X86_64:
+                return "macosx_10_6_x86_64"
+            else:
+                raise OkonomiyakiError(msg.format(self.platform))
+        elif self.platform.os == LINUX:
+            if self.platform.arch.name == X86:
+                return "linux_i686"
+            elif self.platform.arch.name == X86_64:
+                return "linux_x86_64"
+            else:
+                raise OkonomiyakiError(msg.format(self.platform))
+        elif self.platform.family == WINDOWS:
+            if self.platform.arch.name == X86:
+                return "win32"
+            elif self.platform.arch.name == X86_64:
+                return "win_amd64"
+            else:
+                raise OkonomiyakiError(msg.format(self.platform))
         else:
-            return "64"
+            raise OkonomiyakiError(msg.format(self.platform))
+
+    @property
+    def platform_name(self):
+        os = self.platform.os
+        if os == WINDOWS:
+            return "win"
+        elif os == DARWIN:
+            return "osx"
+        elif os == LINUX:
+            family = self.platform.family
+            release = self.platform.release
+            if family == RHEL:
+                parts = release.split(".")
+                if parts[0] == "3":
+                    base = "rh3"
+                elif parts[0] == "5":
+                    base = "rh5"
+                elif parts[0] == "6":
```

Comment:
Above we seem to claim that `rh7` is supported (line 146 `parts[0] in ("3", "5", "6", "7")`). Should that be reflected here as well?